doc/TOR.md

```diff
@@ -1,144 +1,302 @@
+9735 port.
 
---announce-addr=autotor:<torservice_ip:port> : try to generate an temp V2 onion addr.
+#### Creation of an hidden service for a persistent .onion address
 
-NOTE if --always-use-proxy set all traffic will be rooted over the proxy, or if no non-TOR addresses are announced.
+To have a persistent .onion address at which other nodes can connect, it 
+is necessary to set up a [TOR Hidden Service].
 
-you can also set a fixed announce onion addr by option
---announce-addr=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.onion[:port]  (V2 or V3 is allowed)
+*NOTE:In the case only non persistent addresses are required,  
+you don't have to create the hidden service and you can skip this part.*
 
-this addr can be created
-in /etc/tor/torrc
+To do that we will add these lines in the `/etc/tor/torrc`file:
 
-HiddenServiceDir /var/lib/tor/bitcoin-service_v2/
-HiddenServiceVersion 2
+````
+HiddenServiceDir /var/lib/tor/lightningd-service_v2/
 HiddenServicePort 1234 127.0.0.1:9735
-
-and in later v3 enabled tor version's you can use for V3
-
-HiddenServiceDir /var/lib/tor/other_hidden_service_v3/
+````
+If we want to create a version 3 address, we will add also `HiddenServiceVersion 3` so
+the whole section will be:
+````
+HiddenServiceDir /var/lib/tor/lightningd-service_v3/
 HiddenServiceVersion 3
 HiddenServicePort 1234 127.0.0.1:9735
+````
 
+The hidden lightning service  will be reachable at port 1234 (global port)
+of the .onion address, which will be created at the restart of the 
+TOR service.
 
-the addr for
-the --announce-addr option
-or legacy use
+Of course it is possible create a version 2 AND a version 3 address for the 
+same node.
 
-you find after /etc/init.d/tor restart
+Save the file and restart the TOR service. In linux:
 
-i.e.
-in /var/lib/tor/other_hidden_service_v3/hostname
+`/etc/init.d/tor restart` or `sudo systemctl start tor` depending 
+on the configuration of your system.
+
+You will find the newly created address with:
+
+`sudo cat /var/lib/tor/var/lib/tor/lightningd-service_v2/hostname` or
+
+`sudo cat /var/lib/tor/var/lib/tor/lightningd-service_v3/hostname` in the 
+case of a version 3 TOR address.
+
+Now we are able to create:
+
+* Non persistent version 2 .onion address via auto service (NPer.V2)
+
+* Persistent version 2 and version 3 .onion addresseses (Per.V2 e Per.V3). 
+
+Let's see how to use them.
+
+### What do we support
+
+| Case #  | IP Number     | TOR address               |
+| ------- | ------------- | ------------------------- |
+| 1       | Public        | NO                        |
+| 2       | Public        | Pers.V2 [1]               |
+| 3       | Public        | NPers.v2 [2]              |
+| 4       | Not Announced | Pers.V2                   |
+| 5       | Not Announced | NPers.v2                  |
+| 6       | Public        | Pers.V3+NPers.V2          |
+| 7       | Not Announced | Pers.V3+Pers.V2+NPers.V2  |
+| 8       | Public        | NO                        |
```

Comment:
Case 1 and case 8 seems the same.

Also maybe rename Pers.v2 (also called Per.V2 above) to 'v2' and NPers.v2 to 'temp-v2'?

Reply:
Maybe have columns 'Incoming Tor | Outgoing Tor'?  (BTW, torproject.org says 'even though it originally came from an acronym, Tor is not spelled "TOR"')

Reply:
Will do. I have opted for Non-pers.V2 to avoid any possible confusion between non-persistent and temp.

If I have understood correctly in the 8th case there is no Tor connection, just socks5 proxy but feel free to amend it. That specific part is the most tricky for me.

Reply:
> Maybe have columns 'Incoming Tor | Outgoing Tor'?

Done. It is a little incomplete because, if I understood correctly, every "Incoming" use case can also make outgoing Tor connection if the `--proxy=127.0.0.1:9050` option is added. I have tried to outline it in the notes and in the description of the case 1.